Preferences to enable/disable symbol provider

With large file like nasa.xml, the vscode outline doesn't work. It seems lsp4j doesn't support a big symbol provider:

```
[Trace - 17:48:08] Received response 'textDocument/documentSymbol - (1)' in 6234ms. Request failed: Internal error. (-32603).
[Error - 17:48:08] Request textDocument/documentSymbol failed.
  Message: Internal error.
  Code: -32603 
java.util.concurrent.CompletionException: java.lang.OutOfMemoryError: Java heap space
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:273)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:280)
	at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:659)
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:632)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:474)
	at java.util.concurrent.CompletableFuture.postFire(CompletableFuture.java:561)
	at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:580)
	at java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:443)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
Caused by: java.lang.OutOfMemoryError: Java heap space
	at java.lang.StringCoding$StringEncoder.encode(StringCoding.java:300)
	at java.lang.StringCoding.encode(StringCoding.java:344)
	at java.lang.String.getBytes(String.java:918)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageConsumer.consume(StreamMessageConsumer.java:60)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.lambda$handleRequest$1(RemoteEndpoint.java:281)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint$$Lambda$17/1028064261.accept(Unknown Source)
	at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:656)
	... 9 more
```

To fix this problem, we need to provide a preferences to enable/disable symbol provider.

In the future we could provide a new preferences to set the number of symbol returned by the symbol provider. I started to do that in https://github.com/angelozerr/lsp4xml/blob/7bca4bf80031cbcd522eaf41f3ba217ceb942bfb/org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/XMLSymbolsProvider.java#L67

--- a/org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/XMLLanguageServer.java
+++ b/org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/XMLLanguageServer.java
@@ -49,6 +49,7 @@ import org.eclipse.lsp4xml.settings.SharedSettings;
 import org.eclipse.lsp4xml.settings.XMLExperimentalCapabilities;
 import org.eclipse.lsp4xml.settings.XMLFormattingOptions;
 import org.eclipse.lsp4xml.settings.XMLGeneralClientSettings;
+import org.eclipse.lsp4xml.settings.XMLSymbolSettings;
 import org.eclipse.lsp4xml.settings.capabilities.ServerCapabilitiesInitializer;
 import org.eclipse.lsp4xml.settings.capabilities.XMLCapabilityManager;
 import org.eclipse.lsp4xml.utils.FilesUtils;
@@ -137,6 +138,11 @@ public class XMLLanguageServer
 			CompletionSettings newCompletions = xmlClientSettings.getCompletion();
 			if (newCompletions != null) {
 				xmlTextDocumentService.updateCompletionSettings(newCompletions);
+			}
+
+			XMLSymbolSettings newSymbols = xmlClientSettings.getSymbols();
+			if(newSymbols != null) {
+				xmlTextDocumentService.updateSymbolSettings(newSymbols);
 			}
 
 			ServerSettings serverSettings = xmlClientSettings.getServer();

--- a/org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/XMLTextDocumentService.java
+++ b/org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/XMLTextDocumentService.java
@@ -15,6 +15,7 @@ import static org.eclipse.lsp4j.jsonrpc.CompletableFutures.computeAsync;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
 import java.util.List;
 import java.util.concurrent.CancellationException;
 import java.util.concurrent.CompletableFuture;
@@ -73,6 +74,7 @@ import org.eclipse.lsp4xml.services.extensions.CompletionSettings;
 import org.eclipse.lsp4xml.services.extensions.save.AbstractSaveContext;
 import org.eclipse.lsp4xml.settings.SharedSettings;
 import org.eclipse.lsp4xml.settings.XMLFormattingOptions;
+import org.eclipse.lsp4xml.settings.XMLSymbolSettings;
 
 /**
  * XML text document service.
@@ -215,6 +217,11 @@ public class XMLTextDocumentService implements TextDocumentService {
 	@Override
 	public CompletableFuture<List<Either<SymbolInformation, DocumentSymbol>>> documentSymbol(
 			DocumentSymbolParams params) {
+		
+		if(!sharedSettings.symbolSettings.isEnabled()) {
+			return CompletableFuture.completedFuture(Collections.emptyList());
+		}
+
 		return computeAsync((monitor) -> {
 			TextDocument document = getDocument(params.getTextDocument().getUri());
 			DOMDocument xmlDocument = getXMLDocument(document);
@@ -429,6 +436,14 @@ public class XMLTextDocumentService implements TextDocumentService {
 
 	public void updateCompletionSettings(CompletionSettings newCompletion) {
 		sharedSettings.completionSettings.setAutoCloseTags(newCompletion.isAutoCloseTags());
+	}
+
+	public void updateSymbolSettings(XMLSymbolSettings newSettings) {
+		sharedSettings.symbolSettings.setEnabled(newSettings.isEnabled());
+	}
+
+	public XMLSymbolSettings getSharedSymbolSettings() {
+		return sharedSettings.symbolSettings;
 	}
 
 	public boolean isIncrementalSupport() {

--- a/org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/settings/SharedSettings.java
+++ b/org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/settings/SharedSettings.java
@@ -24,12 +24,14 @@ public class SharedSettings {
 	public final FoldingRangeCapabilities foldingSettings;
 	public XMLFormattingOptions formattingSettings;
 	public final XMLValidationSettings validationSettings;
+	public final XMLSymbolSettings symbolSettings;
 
 	public SharedSettings() {
 		this.completionSettings = new CompletionSettings();
 		this.foldingSettings = new FoldingRangeCapabilities();
 		this.formattingSettings = new XMLFormattingOptions(true);
 		this.validationSettings = new XMLValidationSettings();
+		this.symbolSettings = new XMLSymbolSettings();
 	}
 
 	public void setFormattingSettings(XMLFormattingOptions formattingOptions) {

--- a/org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/settings/XMLGeneralClientSettings.java
+++ b/org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/settings/XMLGeneralClientSettings.java
@@ -44,8 +44,18 @@ public class XMLGeneralClientSettings {
 
 	private ServerSettings server;
 
+	private XMLSymbolSettings symbols;
+
 	public void setLogs(LogsSettings logs) {
 		this.logs = logs;
+	}
+
+	public XMLSymbolSettings getSymbols() {
+		return symbols;
+	}
+
+	public void setSymbols(XMLSymbolSettings symbols) {
+		this.symbols = symbols;
 	}
 
 	public LogsSettings getLogs() {

--- a/org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/settings/XMLSymbolSettings.java
+++ b/org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/settings/XMLSymbolSettings.java
@@ -1,0 +1,27 @@
+/*******************************************************************************
+* Copyright (c) 2019 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+
+package org.eclipse.lsp4xml.settings;
+
+/**
+ * XMLSymbolPreferences for Document Symbols
+ */
+public class XMLSymbolSettings {
+
+	private boolean enabled = true;
+
+	public boolean isEnabled() {
+		return enabled;
+	}
+
+	public void setEnabled(boolean enabled) {
+		this.enabled = enabled;
+	}
+}

--- a/org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/settings/capabilities/ClientCapabilitiesWrapper.java
+++ b/org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/settings/capabilities/ClientCapabilitiesWrapper.java
@@ -68,7 +68,7 @@ public class ClientCapabilitiesWrapper {
 		return v3Supported && isDynamicRegistrationSupported(getTextDocument().getRename());
 	}
 
-	public boolean isDocumentSymbolDynamicRegistered() {
+	public boolean isDocumentSymbolDynamicRegistrationSupported() {
 		return v3Supported && isDynamicRegistrationSupported(getTextDocument().getDocumentSymbol());
 	}
 

--- a/org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/settings/capabilities/ServerCapabilitiesInitializer.java
+++ b/org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/settings/capabilities/ServerCapabilitiesInitializer.java
@@ -37,19 +37,14 @@ public class ServerCapabilitiesInitializer {
 	public static ServerCapabilities getNonDynamicServerCapabilities(ClientCapabilitiesWrapper clientCapabilities,
 			boolean isIncremental) {
 		ServerCapabilities serverCapabilities = new ServerCapabilities();
-
+		// @formatter:off
 		serverCapabilities.setTextDocumentSync(DEFAULT_SYNC_OPTION);
-
-		serverCapabilities
-				.setTextDocumentSync(isIncremental ? TextDocumentSyncKind.Incremental : TextDocumentSyncKind.Full);
-
-		serverCapabilities.setDocumentSymbolProvider(!clientCapabilities.isDocumentSymbolDynamicRegistered());
+		serverCapabilities.setTextDocumentSync(isIncremental ? TextDocumentSyncKind.Incremental : TextDocumentSyncKind.Full);
+		serverCapabilities.setDocumentSymbolProvider(!clientCapabilities.isDocumentSymbolDynamicRegistrationSupported());
 		serverCapabilities.setDocumentHighlightProvider(!clientCapabilities.isDocumentHighlightDynamicRegistered());
 		serverCapabilities.setCodeActionProvider(!clientCapabilities.isCodeActionDynamicRegistered());
-		serverCapabilities
-				.setDocumentFormattingProvider(!clientCapabilities.isFormattingDynamicRegistrationSupported());
-		serverCapabilities.setDocumentRangeFormattingProvider(
-				!clientCapabilities.isRangeFormattingDynamicRegistrationSupported());
+		serverCapabilities.setDocumentFormattingProvider(!clientCapabilities.isFormattingDynamicRegistrationSupported());
+		serverCapabilities.setDocumentRangeFormattingProvider(!clientCapabilities.isRangeFormattingDynamicRegistrationSupported());
 		serverCapabilities.setHoverProvider(!clientCapabilities.isHoverDynamicRegistered());
 		serverCapabilities.setRenameProvider(!clientCapabilities.isRenameDynamicRegistrationSupported());
 		serverCapabilities.setFoldingRangeProvider(!clientCapabilities.isRangeFoldingDynamicRegistrationSupported());

--- a/org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/settings/capabilities/XMLCapabilityManager.java
+++ b/org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/settings/capabilities/XMLCapabilityManager.java
@@ -45,6 +45,7 @@ import org.eclipse.lsp4j.UnregistrationParams;
 import org.eclipse.lsp4j.services.LanguageClient;
 import org.eclipse.lsp4xml.XMLTextDocumentService;
 import org.eclipse.lsp4xml.settings.XMLFormattingOptions;
+import org.eclipse.lsp4xml.settings.XMLSymbolSettings;
 
 /**
  * Manager for capability related tasks
@@ -125,9 +126,6 @@ public class XMLCapabilityManager {
 		if (this.getClientCapabilities().isDocumentHighlightDynamicRegistered()) {
 			registerCapability(DOCUMENT_HIGHLIGHT_ID, TEXT_DOCUMENT_HIGHLIGHT);
 		}
-		if (this.getClientCapabilities().isDocumentSymbolDynamicRegistered()) {
-			registerCapability(DOCUMENT_SYMBOL_ID, TEXT_DOCUMENT_DOCUMENT_SYMBOL);
-		}
 		if (this.getClientCapabilities().isRangeFoldingDynamicRegistrationSupported()) {
 			registerCapability(FOLDING_RANGE_ID, TEXT_DOCUMENT_FOLDING_RANGE);
 		}
@@ -164,6 +162,13 @@ public class XMLCapabilityManager {
 		if (this.getClientCapabilities().isRangeFormattingDynamicRegistrationSupported()) {
 			toggleCapability(formattingPreferences.isEnabled(), FORMATTING_RANGE_ID,
 					ServerCapabilitiesConstants.TEXT_DOCUMENT_RANGE_FORMATTING, null);
+		}
+
+		XMLSymbolSettings symbolSettings = this.textDocumentService.getSharedSymbolSettings();
+
+		if (this.getClientCapabilities().isDocumentSymbolDynamicRegistrationSupported()) {
+			toggleCapability(symbolSettings.isEnabled(), DOCUMENT_SYMBOL_ID,
+					TEXT_DOCUMENT_DOCUMENT_SYMBOL, null);
 		}
 	}
 

--- a/org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/settings/SettingsTest.java
+++ b/org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/settings/SettingsTest.java
@@ -10,22 +10,25 @@
  */
 package org.eclipse.lsp4xml.settings;
 
-import org.eclipse.lsp4j.FormattingOptions;
-import org.eclipse.lsp4j.InitializeParams;
-import org.eclipse.lsp4xml.XMLLanguageServer;
-import org.eclipse.lsp4xml.extensions.contentmodel.settings.ContentModelSettings;
-import org.eclipse.lsp4xml.utils.FilesUtils;
-import org.junit.After;
-import org.junit.Assert;
-import org.junit.Test;
-
+import static java.io.File.separator;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
 
 import java.io.File;
 
 import com.google.gson.Gson;
 import com.google.gson.JsonObject;
-import static java.io.File.separator;
+
+import org.eclipse.lsp4j.FormattingOptions;
+import org.eclipse.lsp4j.InitializeParams;
+import org.eclipse.lsp4xml.XMLLanguageServer;
+import org.eclipse.lsp4xml.extensions.contentmodel.settings.ContentModelSettings;
+import org.eclipse.lsp4xml.utils.FilesUtils;
+import org.junit.After;
+import org.junit.Test;
 
 /**
  * Tests for settings.
@@ -33,56 +36,60 @@ import static java.io.File.separator;
 public class SettingsTest {
 
 	private static String testFolder = "TestXMLCacheFolder";
-	private static String targetTestFolder = "target" + separator +"generated-test-sources";
-	
-	
+	private static String targetTestFolder = "target" + separator + "generated-test-sources";
+
 	@After
 	public void cleanup() {
 		String path = System.getProperty("user.dir") + separator + targetTestFolder + separator + testFolder;
-		
+
 		File f = new File(path);
 		if (f.exists()) {
 			f.delete();
 		}
 	}
-
-	private final String json = "{\r\n" + //
-			"	\"settings\": {\r\n" + //
-			// Content model settings
-			"		\"xml\": {\r\n" + 
-			"			\"fileAssociations\": [\r\n" + //
-			"				{\r\n" + //
-			"					\"systemId\": \"src\\\\test\\\\resources\\\\xsd\\\\spring-beans-3.0.xsd\",\r\n" + //
-			"					\"pattern\": \"**/test*.xml\"\r\n" + //
-			"				},\r\n" + //
-			"				{\r\n" + //
-			"					\"systemId\": \"src\\\\test\\\\resources\\\\xsd\\\\projectDescription.xsd\",\r\n" + //
-			"					\"pattern\": \"projectDescription.xml\"\r\n" + //
-			"				}\r\n" + //
-			"			],\r\n" + //
-			"			\"catalogs\": [\r\n" + //
-			"				\"src\\\\test\\\\resources\\\\catalogs\\\\catalog.xml\"\r\n" + //
-			"			],\r\n" + //
-			"			\"validation\": {\r\n" + //
-			"				\"enabled\": true,\r\n" + //
-			"				\"schema\": false\r\n" + //
-			"			},\r\n" + //
-			// Client (commons) settings
-			"			\"format\": {\r\n" + //
-			"				\"tabSize\": 10,\r\n" + //
-			"				\"insertSpaces\": false,\r\n" + //
-			"				\"splitAttributes\": true,\r\n" + //
-			"				\"joinCDATALines\": true,\r\n" + //
-			"				\"formatComments\": true,\r\n" + //
-			"				\"joinCommentLines\": true,\r\n" + //
-			"				\"quotations\": " + XMLFormattingOptions.DOUBLE_QUOTES_VALUE + "\r\n" + //
-			"			},\r\n" + 
-			"			\"server\": {\r\n" + //
-			"				\"workDir\": \"~/" + testFolder + "/Nested\"\r\n" + //
-			"			}\r\n" + 
-			"		}\r\n" + 
-			"	}\r\n" + 
-			"}";
+	// @formatter:off
+	private final String json = 
+	"{\r\n" +
+	"	\"settings\": {\r\n" + //
+	// Content model settings
+	"		\"xml\": {\r\n" + 
+	"			\"fileAssociations\": [\r\n" + //
+	"				{\r\n" + //
+	"					\"systemId\": \"src\\\\test\\\\resources\\\\xsd\\\\spring-beans-3.0.xsd\",\r\n" + //
+	"					\"pattern\": \"**/test*.xml\"\r\n" + //
+	"				},\r\n" + //
+	"				{\r\n" + //
+	"					\"systemId\": \"src\\\\test\\\\resources\\\\xsd\\\\projectDescription.xsd\",\r\n" + //
+	"					\"pattern\": \"projectDescription.xml\"\r\n" + //
+	"				}\r\n" + //
+	"			],\r\n" + //
+	"			\"catalogs\": [\r\n" + //
+	"				\"src\\\\test\\\\resources\\\\catalogs\\\\catalog.xml\"\r\n" + //
+	"			],\r\n" + //
+	"			\"validation\": {\r\n" + //
+	"				\"enabled\": true,\r\n" + //
+	"				\"schema\": false\r\n" + //
+	"			},\r\n" + //
+	// Client (commons) settings
+	"			\"format\": {\r\n" + //
+	"				\"tabSize\": 10,\r\n" + //
+	"				\"insertSpaces\": false,\r\n" + //
+	"				\"splitAttributes\": true,\r\n" + //
+	"				\"joinCDATALines\": true,\r\n" + //
+	"				\"formatComments\": true,\r\n" + //
+	"				\"joinCommentLines\": true,\r\n" + //
+	"				\"quotations\": " + XMLFormattingOptions.DOUBLE_QUOTES_VALUE + "\r\n" + //
+	"			},\r\n" + 
+	"			\"server\": {\r\n" + //
+	"				\"workDir\": \"~/" + testFolder + "/Nested\"\r\n" + //
+	"			},\r\n" + 
+	"			\"symbols\": {\r\n" + //
+	"				\"enabled\": true\r\n" + //
+	"			}\r\n" + 
+	"		}\r\n" + 
+	"	}\r\n" + 
+	"}";
+	// @formatter:on
 
 	@Test
 	public void initializationOptionsSettings() {
@@ -96,27 +103,30 @@ public class SettingsTest {
 		// Test client commons settings
 		initializationOptionsSettings = AllXMLSettings.getAllXMLSettings(initializationOptionsSettings);
 		XMLGeneralClientSettings settings = XMLGeneralClientSettings.getGeneralXMLSettings(initializationOptionsSettings);
-		Assert.assertNotNull(settings);
+		assertNotNull(settings);
 		// Server
-		Assert.assertEquals("~/" + testFolder + "/Nested", settings.getServer().getWorkDir());
+		assertEquals("~/" + testFolder + "/Nested", settings.getServer().getWorkDir());
 
 		// Test content model extension settings
 		ContentModelSettings cmSettings = ContentModelSettings.getContentModelXMLSettings(initializationOptionsSettings);
-		Assert.assertNotNull(cmSettings);
+		assertNotNull(cmSettings);
 		// Catalog
-		Assert.assertNotNull(cmSettings.getCatalogs());
-		Assert.assertEquals(1, cmSettings.getCatalogs().length);
-		Assert.assertEquals("src\\test\\resources\\catalogs\\catalog.xml", cmSettings.getCatalogs()[0]);
+		assertNotNull(cmSettings.getCatalogs());
+		assertEquals(1, cmSettings.getCatalogs().length);
+		assertEquals("src\\test\\resources\\catalogs\\catalog.xml", cmSettings.getCatalogs()[0]);
 		// File associations
-		Assert.assertNotNull(cmSettings.getFileAssociations());
-		Assert.assertEquals(2, cmSettings.getFileAssociations().length);
-		Assert.assertEquals("src\\test\\resources\\xsd\\spring-beans-3.0.xsd",
-				cmSettings.getFileAssociations()[0].getSystemId());
-		Assert.assertEquals("**/test*.xml", cmSettings.getFileAssociations()[0].getPattern());
+		assertNotNull(cmSettings.getFileAssociations());
+		assertEquals(2, cmSettings.getFileAssociations().length);
+		assertEquals("src\\test\\resources\\xsd\\spring-beans-3.0.xsd", cmSettings.getFileAssociations()[0].getSystemId());
+		assertEquals("**/test*.xml", cmSettings.getFileAssociations()[0].getPattern());
 		// Diagnostics
-		Assert.assertNotNull(cmSettings.getValidation());
-		Assert.assertEquals(true, cmSettings.getValidation().isEnabled());
-		Assert.assertEquals(false, cmSettings.getValidation().isSchema());
+		assertNotNull(cmSettings.getValidation());
+		assertEquals(true, cmSettings.getValidation().isEnabled());
+		assertEquals(false, cmSettings.getValidation().isSchema());
+		// Symbols
+		assertNotNull(settings.getSymbols());
+		assertEquals(true, settings.getSymbols().isEnabled());
+
 	}
 
 	private static InitializeParams createInitializeParams(String json) {
@@ -143,27 +153,27 @@ public class SettingsTest {
 
 		xmlFormattingOptions.setQuotations("InvalidValue"); // set a value that is not recognized
 
-		Assert.assertEquals(5, xmlFormattingOptions.getTabSize()); // value coming from the request formattingOptions
-		Assert.assertFalse(xmlFormattingOptions.isInsertSpaces()); // formattingOptions doesn't defines insert spaces
+		assertEquals(5, xmlFormattingOptions.getTabSize()); // value coming from the request formattingOptions
+		assertFalse(xmlFormattingOptions.isInsertSpaces()); // formattingOptions doesn't defines insert spaces
 		// flag
 
-		Assert.assertFalse(xmlFormattingOptions.isJoinCommentLines());// Since default for JoinCommentLines is False
+		assertFalse(xmlFormattingOptions.isJoinCommentLines());// Since default for JoinCommentLines is False
 
 		// Assumes the "InvalidValue" will be overridden
-		Assert.assertTrue(xmlFormattingOptions.getQuotations().equals(XMLFormattingOptions.DOUBLE_QUOTES_VALUE));
+		assertTrue(xmlFormattingOptions.getQuotations().equals(XMLFormattingOptions.DOUBLE_QUOTES_VALUE));
 
 		// merge with shared sharedXMLFormattingOptions (formatting settings created in
 		// the InitializeParams
 		xmlFormattingOptions.merge(sharedXMLFormattingOptions);
-		Assert.assertEquals(5, xmlFormattingOptions.getTabSize()); // tab size is kept as 5 (and not updated with
+		assertEquals(5, xmlFormattingOptions.getTabSize()); // tab size is kept as 5 (and not updated with
 		// shared value 10), because only the request's
 		// formattingOptions object is allowed to define it.
-		Assert.assertFalse(xmlFormattingOptions.isInsertSpaces()); // insert spaces is kept as false because only the
-																																// request's
+		assertFalse(xmlFormattingOptions.isInsertSpaces()); // insert spaces is kept as false because only the
+																												// request's
 		// formattingOptions object is allowed to define it.
-		Assert.assertTrue(xmlFormattingOptions.isJoinCommentLines());
+		assertTrue(xmlFormattingOptions.isJoinCommentLines());
 
-		Assert.assertTrue(xmlFormattingOptions.getQuotations().equals(XMLFormattingOptions.SINGLE_QUOTES_VALUE));
+		assertTrue(xmlFormattingOptions.getQuotations().equals(XMLFormattingOptions.SINGLE_QUOTES_VALUE));
 	}
 
 	@Test
@@ -171,7 +181,7 @@ public class SettingsTest {
 		// Emulate InitializeParams#getInitializationOptions() object created as
 		// JSONObject when XMLLanguageServer#initialize(InitializeParams params) is
 		// called
-		
+
 		InitializeParams params = createInitializeParams(json);
 		Object initializationOptionsSettings = InitializationOptionsSettings.getSettings(params);
 		XMLLanguageServer languageServer = new XMLLanguageServer();
@@ -180,11 +190,12 @@ public class SettingsTest {
 		String userDir = System.getProperty("user.dir");
 		try {
 			System.setProperty("user.home", userDir + separator + targetTestFolder); // .../org.eclipse.lsp4xml/target/generated-test-sources/
-		
+
 			languageServer.updateSettings(initializationOptionsSettings);
 
-			//Ensure the expanded absolute path is being used.
-			Assert.assertEquals(System.getProperty("user.home") + separator + testFolder + separator + "Nested", FilesUtils.getCachePathSetting());
+			// Ensure the expanded absolute path is being used.
+			assertEquals(System.getProperty("user.home") + separator + testFolder + separator + "Nested",
+					FilesUtils.getCachePathSetting());
 		} catch (Exception e) {
 			fail();
 		} finally {

